src/NuGet.Core/NuGet.ProjectManagement/LoggerAdapter.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using NuGet.Logging;
+
+namespace NuGet.ProjectManagement
+{
+    /// <summary>
+    /// ILogger -> INuGetProjectContext
+    /// </summary>
+    public class LoggerAdapter : ILogger
+    {
+        public INuGetProjectContext ProjectLogger { get; }
```

Comment:
It allows either one to be used.

Ideally INuGetProjectContext should extend ILogger and the old INuGetProjectContext  logging format converted over and removed from the product.